### DISCUSSION TITLE
[ROS2]Actual Robot (HSRC)

### DISCUSSION THREAD

**Discussion by okdhryk** [HumanSupportRobot/QandA#60] [Q&A]
The installation is proceeding according to this page.
https://github.com/hsr-project/hsr_ros2_doc

Actual Robot
...
...
rosdep install --from-paths . -y --ignore-src

I get the following error.

~/hsr_ros2_ws$ rosdep install --from-paths . -y --ignore-src
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
openni2_camera: Cannot locate rosdep definition for [camera_info_manager]
hsrb_base_controllers: Cannot locate rosdep definition for [realtime_tools]
hsrb_bringup: Cannot locate rosdep definition for [usb_cam]
hsrb_robot_hardware: Cannot locate rosdep definition for [transmission_interface]
hsrb_gripper_controller: Cannot locate rosdep definition for [joint_trajectory_controller]
hsrb_description: Cannot locate rosdep definition for [xacro]

**Comment by yoshida6298** [comment]
I think it is due to rosdep not working properly due to end of life of foxy. You can build by manually installing the dependent package corresponding to each key.

**Comment by okdhryk** [comment]
I build by manually installing the dependent package.

Errors still remain ...
Has anyone else had success with this?
> CMake Error at CMakeLists.txt:73 (rosidl_get_typesupport_target):
>   Unknown CMake command "rosidl_get_typesupport_target".


$ rosdep install --from-paths . -y --ignore-src
#All required rosdeps installed successfully

$ colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
Starting >>> hsrb_servomotor_protocol
Starting >>> tmc_control_msgs
Starting >>> hsrb_base_controllers
Starting >>> openni2_camera
Starting >>> exxx_control_table
Starting >>> hsrb_description
Starting >>> hsrb_meshes
Finished <<< exxx_control_table [1.07s]
Finished <<< hsrb_meshes [1.12s]
Finished <<< hsrb_description [1.21s]
Finished <<< hsrb_servomotor_protocol [3.96s]
Starting >>> hsrb_robot_hardware
--- stderr: openni2_camera
CMake Error at CMakeLists.txt:73 (rosidl_get_typesupport_target):
  Unknown CMake command "rosidl_get_typesupport_target".


---
Failed   <<< openni2_camera [4.29s, exited with code 1]
Aborted  <<< tmc_control_msgs [4.56s]
Aborted  <<< hsrb_base_controllers [1min 5s]
Aborted  <<< hsrb_robot_hardware [1min 19s]

Summary: 4 packages finished [1min 23s]
  1 package failed: openni2_camera
  3 packages aborted: hsrb_base_controllers hsrb_robot_hardware tmc_control_msgs
  1 package had stderr output: openni2_camera
  2 packages not processed

**Comment by k-takeshita** [comment]
Is it the same issue as this link?
https://github.com/ros-drivers/openni2_camera/issues/130
